### PR TITLE
Ledger monday updates

### PR DESCRIPTION
**Pull request by mrose17**
auditor: @diracdeltas 

1. build for node-anonize2-relic-emscripten to work with `asar` packaging
2. poll for unconfirmed balance and better synchornization
3. use `clientOptions.verboseP` for some calls to `console.log`
4. call `updatePublisherInfo` at the correct places

### CODE REVIEW

File: app/ledger.js [hunk 1]
```diff
@@ -116,8 +117,14 @@ var boot = () => {
 
     if (err.code !== 'ENOENT') console.log('statePath read error: ' + err.toString())
 
-    client = (require('ledger-client'))(null, underscore.extend(clientOptions, { roundtrip: roundtrip }), null)
+    try {
+      client = (require('ledger-client'))(null, underscore.extend({ roundtrip: roundtrip }, clientOptions), null)
+    } catch (ex) {
+      bootP = false
+      return console.log('ledger-client error: ' + ex.toString() + '\n' + ex.stack)
```
Comment (diracdeltas) [inline app/ledger.js]: this now throws:
```
ledger-client error: Error: ENOENT: no such file or directory, open '/Users/yan/repos/browser-laptop/Brave-darwin-x64/Brave.app/Contents/Resources/app.asar.unpacked/node_modules/node-anonize2-relic-emscripten/anonize2.js.mem'
Error: ENOENT: no such file or directory, open '/Users/yan/repos/browser-laptop/Brave-darwin-x64/Brave.app/Contents/Resources/app.asar.unpacked/node_modules/node-anonize2-relic-emscripten/anonize2.js.mem'
    at Error (native)
    at Object.fs.openSync (fs.js:640:18)
    at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:167:20)
    at Object.fs.readFileSync (fs.js:508:33)
    at Object.fs.readFileSync (ELECTRON_ASAR.js:500:29)
    at Object.read (/Users/yan/repos/browser-laptop/Brave-darwin-x64/Brave.app/Contents/Resources/app.asar/node_modules/node-anonize2-relic-emscripten/anonize2.js:18:1355)
    at Object.readBinary (/Users/yan/repos/browser-laptop/Brave-darwin-x64/Brave.app/Contents/Resources/app.asar/node_modules/node-anonize2-relic-emscripten/anonize2.js:18:1623)
    at Object.<anonymous> (/Users/yan/repos/browser-laptop/Brave-darwin-x64/Brave.app/Contents/Resources/app.asar/node_modules/node-anonize2-relic-emscripten/anonize2.js:33:3553)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
```

File: app/ledger.js [hunk 2]
```diff
@@ -899,6 +922,29 @@ var getStateInfo = (state) => {
   updateLedgerInfo()
 }
 
+var getBalance = () => {
+  setTimeout(getBalance, msecs.minute)
```
Comment (diracdeltas) [inline app/ledger.js]: this timeout should be cleared when ledger is disabled

File: app/ledger.js (hunk 1 above)
Comment (diracdeltas) [inline app/ledger.js]: `Brave-darwin-x64/Brave.app/Contents/Resources/app.asar.unpacked/node_modules/node-anonize2-relic-emscripten/` exists but is empty

File: tools/buildPackage.js [hunk 3]
```diff
@@ -122,4 +122,8 @@ if (isLinux) {
 
 cmds.push('mkdirp ' + path.join(buildDir, 'resources', 'app.asar.unpacked', 'node_modules', 'spellchecker', 'vendor', 'hunspell_dictionaries'))
 cmds.push('ncp ' + path.join('node_modules', 'spellchecker', 'vendor', 'hunspell_dictionaries') + ' ' + path.join(buildDir, 'resources', 'app.asar.unpacked', 'node_modules', 'spellchecker', 'vendor', 'hunspell_dictionaries'))
+
+cmds.push('mkdirp ' + path.join(buildDir, 'Brave.app', 'Contents', 'Resources', 'app.asar.unpacked', 'node_modules', 'node-anonize2-relic-emscripten'))
+cmds.push('ncp ' + path.join('node_modules', 'node-anonize2-relic-emscripten', 'anonize2.js.mem') + ' ' + path.join(buildDir, 'Brave.app', 'Contents', 'Resources', 'app.asar.unpacked', 'node_modules', 'node-anonize2-relic-emscripten'))
```
Comment (diracdeltas) [inline tools/buildPackage.js]: error above appears to be fixed by changing this line to
```
cmds.push('ncp ' + path.join('node_modules', 'node-anonize2-relic-emscripten', 'anonize2.js.mem') + ' ' + path.join(buildDir, 'Brave.app', 'Contents', 'Resources', 'app.asar.unpacked', 'node_modules', 'node-anonize2-relic-emscripten', 'anonize2.js.mem'))
```

**Comment by mrose17** [comment]
OBE by new branch

**Pull request closed without merging**